### DISCUSSION TITLE
ojs_define does not work in webr chunk

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#9119] [Q&A]
### Description

The following code with R and OJS works in a quarto document:

````
```{r}
colnames(iris) <- c("sepal_length", "sepal_width", "petal_length", "petal_width", "species")
ojs_define(ojs_iris = iris)
```

```{ojs}
iris = transpose(ojs_iris)
```

```{ojs}
Plot.dot(iris, {
  x: "petal_length",
  y: "sepal_width",
  stroke: "species",
  channel: {Species: "species"},
  tip: true
}).plot({
  grid: true,
  symbol: { legend: true }
})
```
````

![Screenshot 2024-03-17 at 20 17 20](https://github.com/quarto-dev/quarto-cli/assets/5083500/9151a068-3239-438a-8dfa-e2ad970fe09f)

But, if the first code block is changed from `r` to `webr-r`, I get this error: 

````
```{webr-r}
colnames(iris) <- c("sepal_length", "sepal_width", "petal_length", "petal_width", "species")
ojs_define(ojs_iris = iris)
```
````

![Screenshot 2024-03-17 at 20 18 31](https://github.com/quarto-dev/quarto-cli/assets/5083500/8ddf020d-a4f5-4a20-9040-43a32ad483f2)

`quarto 1.4.549`
`quarto-webr 0.4.1-dev.1`

**Comment by mcanouil** [comment]
Yes, because those two are different runtime environments.
Observable does not really access code content.
Quarto runs the code cells, set Observable environment with the result of code cell.
Then the browser can use the result and execute Observable.

This is not an issue with Quarto.
